src/never.rs

```diff
@@ -0,0 +1,91 @@
+use {Future, Poll};
+
+/// A future which is never resolved.
+///
+/// This future can be created with the `empty` function.
+#[derive(Copy, Clone)]
+pub struct Never {}
+
+/// Creates a future which never resolves, representing a computation that never
+/// finishes.
+///
+/// The returned future will never resolve with a success but is still
+/// susceptible to cancellation. That is, if a callback is scheduled on the
+/// returned future, it is only run once the future is dropped (canceled).
+pub fn never<T, E>() -> impl Future<Item = T, Error = E> + Send + 'static {
```

Comment:
The point of this function was to allow a `Never` to coerce into any `Future<Item = T, Error = E>` rather than a `Future<Item = !, Error = !`. Ideally, `T` and `E` would default to `!` (hence my comment below regarding default type parameters).

With that in mind, this function really is returning a `map_err::MapErr<map::Map<Never, never_id::<T>>, never_id::<E>>`.